src/mentorship/features/kaggle/storesales/etl.py

```diff
@@ -20,4 +32,49 @@ def transform(self, X, y=None):
         X = X.sort_values(by=[self.id_column], ascending=True, ignore_index=True)
         X = X.drop(columns=[self.id_column])
         X['family'] = X['family'].str.lower()
+
+        # lags for train set
+        if self.adding_lags:
+            X_copy = X.copy()
+            if self.use_final_metric:
+                X_copy[self.target_col] = np.log(X_copy[self.target_col] + 1)
+            for current_lag in self.lags:
+                X.loc[:, 'lag_{}'.format(current_lag)] = X_copy.groupby(['store_nbr', 'family'])[self.target_col].shift(
+                    current_lag)
+
+        # rolling features for train set
+        if self.adding_rolling_features:
+            X_copy = X.copy()
+            if self.use_final_metric:
+                X_copy[self.target_col] = np.log(X_copy[self.target_col] + 1)
+
+            grp = X_copy[self.target_col].groupby(X_copy['store_nbr'])
+            for period in self.rolling_days:
+                rolling_slices = [sliding_window_view(v, period) for _, v in grp]
+                for function, aggregate in self.rolling_aggr.items():
+                    # giving correct indices to the pd.Series with rolling values (to correctly assign them to X)
+                    X.loc[period * X['store_nbr'].nunique():, f'rolling_{period}d_{aggregate}'] = pd.Series(
+                        function(rolling_slices, -1).ravel(order='F'),
+                        index=range(period * X['store_nbr'].nunique(), X.shape[0] + X['store_nbr'].nunique()))
         return X, y
+
+    def adding_is_holiday_feature(self, X):
+        X_copy = X.copy()
+        holidays_data = pd.read_csv(DATA_ROOT / 'holidays_events.csv').drop_duplicates(subset='date')
+        X_copy = X_copy.merge(holidays_data, on='date', how='left', indicator=True)
+        X_copy['_merge'] = LabelEncoder().fit_transform(X_copy['_merge'].tolist())
+        return X_copy['_merge']
+
+    def adding_stores_data(self, X, columns_to_add):
```

Comment:
Let's use imperative style for function naming:
```suggestion
    def add_store_features(self, X, columns_to_add):
```